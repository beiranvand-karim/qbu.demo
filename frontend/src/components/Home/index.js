import React, {Component} from 'react'
import {Link} from "react-router-dom"

export class Home extends Component {
  render() {
    return (
      <>
        <h1>home page</h1>
        <Link to="/createQuestion">create a question</Link>
      </>
    )
  }
}

export default Home;
