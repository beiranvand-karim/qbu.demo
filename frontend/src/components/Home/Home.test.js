import {shallow} from 'enzyme'
import React from "react"
import Home from "./"
import renderer from "react-test-renderer"
import {MemoryRouter} from "react-router-dom"

describe('<Home>', () => {

   it('should render without crashing', () => {
      shallow(<Home/>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><Home/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });

});
