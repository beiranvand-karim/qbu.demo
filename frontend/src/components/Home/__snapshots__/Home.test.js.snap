// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> should matches the snapshot 1`] = `
Array [
  <h1>
    home page
  </h1>,
  <a
    href="/createQuestion"
    onClick={[Function]}
  >
    create a question
  </a>,
]
`;
